### PR TITLE
minor: fix  to support scalars

### LINKED ISSUE #8386: [BUG]: windows frame don't support scalars in ORDER BY/PARTITION BY clauses
### Describe the bug

During work on #8371, @comphead proposed that we can test `partition by null` and `order by null`

I test this and find  `partition by null` and `order by null` will fail in all windows function(maybe, I'm not test all)
```
❯ CREATE TABLE t1 (a int) AS VALUES (1), (2), (3);
0 rows in set. Query took 0.018 seconds.

❯ select rank() over (partition by null) from t1;
thread 'tokio-runtime-worker' panicked at /home/hhj/datafusion/datafusion/physical-plan/src/repartition/mod.rs:208:79:
called `Result::unwrap()` on an `Err` value: InvalidArgumentError("must either specify a row count or at least one column")
Execution error: Expects PARTITION BY expression to be ordered

❯ select rank() over (order by null) from t1;
type_coercion
caused by
Internal error: Cannot run range queries on datatype: Null.
This was likely caused by a bug in DataFusion's code and we would welcome that you file an bug report in our issue tracker

❯ select sum(a) over (order by null) from t1;
type_coercion
caused by
Internal error: Cannot run range queries on datatype: Null.
This was likely caused by a bug in DataFusion's code and we would welcome that you file an bug report in our issue tracker

❯ select sum(a) over (partition by null) from t1;
Internal error: All partition by columns should have an ordering.
This was likely caused by a bug in DataFusion's code and we would welcome that you file an bug report in our issue tracker
```

### To Reproduce

_No response_

### Expected behavior

correct work like duckdb

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by comphead**
## Which issue does this PR close?

Closes #8386 .

## Rationale for this change

Support scalars for `PARTITION BY` clause to avoid `Expects PARTITION BY expression to be ordered` error.

## Are these changes tested?
Yes

## Are there any user-facing changes?
No

### CODE REVIEW

**Comment by comphead** [comment]
@mustafasrepo please review

**Review by mustafasrepo — APPROVED** [review]
LGTM! Thanks @comphead. I have one minor comment. However It is not important for this PR.

File: datafusion/sql/src/expr/function.rs [hunk 1]
```diff
@@ -90,6 +90,8 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
             let partition_by = window
                 .partition_by
                 .into_iter()
+                //ignore window spec PARTITION BY for scalar values
+                .filter(|e| !matches!(e, sqlparser::ast::Expr::Value { .. },))
                 .map(|e| self.sql_expr_to_logical_expr(e, schema, planner_context))
                 .collect::<Result<Vec<_>>>()?;
             let mut order_by = self.order_by_to_sort_expr(
```
Comment (mustafasrepo) [inline datafusion/sql/src/expr/function.rs:97]: According to test it seems that we have no problem with ScalarValues inside window `order_by`. However, I think we can apply similar filtering done for `window.partition_by` to the `window.order_by`.

File: datafusion/sql/src/expr/function.rs [hunk 2]
```diff
@@ -90,6 +90,8 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
             let partition_by = window
                 .partition_by
                 .into_iter()
+                //ignore window spec PARTITION BY for scalar values
```
Comment (alamb) [inline datafusion/sql/src/expr/function.rs:93]:
```suggestion
                // ignore window spec PARTITION BY for scalar values
                // as they do not change and thus do not generate new partitions
```

I also double checked https://docs.rs/sqlparser/latest/sqlparser/ast/enum.Value.html is indeed the right node for literals

File: datafusion/sqllogictest/test_files/window.slt [hunk 3]
```diff
@@ -3794,8 +3794,36 @@ select a,
...
 2 1
 
+# support scalar value in ORDER BY 
 query I
 select rank() over (order by 1) rnk from (select 1 a union all select 2 a) x
 ----
 1
 1
+
+# support scalar value in both ORDER BY and PARTITION BY, RANK function
```
Comment (alamb) [inline datafusion/sqllogictest/test_files/window.slt:3804]: 👍  as @mustafasrepo  points out

**Review by alamb — APPROVED** [review]
Looks good to me -- thank you @comphead and @mustafasrepo

**Pull request merged**